node/pkg/watchers/near/watcher.go

```diff
@@ -1,556 +1,309 @@
 package near
 
 import (
-	"bytes"
 	"context"
-	"encoding/hex"
-	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-	"strings"
-	"sync"
 	"time"
 
 	"github.com/certusone/wormhole/node/pkg/common"
 	"github.com/certusone/wormhole/node/pkg/p2p"
 	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
 	"github.com/certusone/wormhole/node/pkg/readiness"
 	"github.com/certusone/wormhole/node/pkg/supervisor"
-	eth_common "github.com/ethereum/go-ethereum/common"
+	"github.com/certusone/wormhole/node/pkg/watchers/near/nearapi"
+	"github.com/certusone/wormhole/node/pkg/watchers/near/timerqueue"
 	"github.com/mr-tron/base58"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
-	"github.com/tidwall/gjson"
 	"github.com/wormhole-foundation/wormhole/sdk/vaa"
 	"go.uber.org/zap"
 )
 
-type (
-	// Watcher is responsible for looking over Near blockchain and reporting new transactions to the wormhole contract
-	Watcher struct {
-		mainnet bool
+var (
 
-		nearRPC          string
-		wormholeContract string
+	// how long to initially wait between observing a transaction and attempting to process the transaction.
+	// To successfully process the transaction, all receipts need to be finalized, which typically only occurs two blocks later or so.
+	// transaction processing will be retried with exponential backoff, i.e. transaction may stay in the queque for ca. initialTxProcDelay^(txProcRetry+2) time.
+	initialTxProcDelay = time.Second * 3
 
-		msgChan  chan *common.MessagePublication
-		obsvReqC chan *gossipv1.ObservationRequest
+	blockPollInterval = time.Millisecond * 200
 
-		next_round  uint64
-		final_round uint64
+	workerCountTxProcessing int  = 10
+	workerChunkFetching     int  = 4                // this value should be set to the amount of chunks in a NEAR block, such that they can all be fetched in parallel
```

Comment:
On average, I am seeing up to 10 chunks in a busy block... so, maybe this should be 10?

Reply:
There is also `nearRPCConcurrentConnections = 10` in the nearapi.go, which we should increase as well if we increase this one. Any insights into how many concurrent HTTP requests the NEAR RPC server might support?

Reply:
I have had 10 concurrent connections in my tests several times but I have always been curious for, at what point, they deny us as a Dos attack.  I have seen a lot of chunks in one block many times where each chunk might only have 1 or 2 txn's.   Its why I would rather bias to more chunk threads.

Reply:
It seems like the NEAR JSON RPC node is starting up with 4 workers (https://github.com/near/nearcore/blob/8dc9a0bab8aa4648fc7af777e9fa7e3e545c95a5/chain/jsonrpc/src/lib.rs#L1372) and actix_web by default supports 256 concurrent TLS connections per worker and 25k non-TLS (https://actix.rs/actix-web/actix_web/struct.HttpServer.html#method.workers). So I think we should be good to at least increase the amount of concurrent connections to 500. And maybe set the `workerChunkFetching` to 20 and `workerCountTxProcessing` to 100 (see my next commit)? 

According to NEAR stats (https://explorer.near.org/stats), the network has max. 2M tx/day (unfortunately they don't give a finer granularity), so this should also be sufficient to support the max NEAR network throughput.